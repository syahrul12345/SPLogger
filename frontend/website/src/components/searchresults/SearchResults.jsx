import React from "react";

import { Grid, Card, CardContent, Typography } from "@material-ui/core";
// Redux
import { connect } from "react-redux";

const SearchResults = (props) => {
  const { searchResults } = props;
  return (
    <>
      <Typography variant="h5"> Search Results</Typography>
      <Grid container spacing={3}>
        {searchResults.map((result) => {
          return (
            <Grid key={result.Brand} item xs={12} md={4}>
              <Card>
                <CardContent>
                  <Typography variant="h5">Brand: {result.Brand}</Typography>
                  <Typography variant="h5">Model: {result.Model}</Typography>
                  <Typography variant="body1">
                    Status: {result.Status}
                  </Typography>
                  <Typography variant="subtitle2">
                    Serial Number: {result.SerialNumber}
                  </Typography>
                  <Typography variant="subtitle2">
                    Date Bought: {result.DateBought}
                  </Typography>
                  <Typography variant="subtitle2">
                    Updated At: {result.CreatedAt}
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
          );
        })}
      </Grid>
    </>
  );
};

const mapStateToProps = (state) => {
  return {
    searchResults: state.user.searchResults,
  };
};

const ConnectedSearchResults = connect(mapStateToProps)(SearchResults);
export { SearchResults, ConnectedSearchResults };
