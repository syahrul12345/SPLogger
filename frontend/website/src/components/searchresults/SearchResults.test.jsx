import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Card, CardContent } from '@material-ui/core';

import { SearchResults } from './SearchResults';

const mockSearchResults = [
  {
      "ID": 22,
      "CreatedAt": "2020-06-26T07:51:48.306436Z",
      "UpdatedAt": "2020-06-26T07:51:48.306436Z",
      "SerialNumber": "CH-1",
      "Brand": "Starbucks",
      "Status": "New",
      "Model": "Cup",
      "DateBought": "Today"
  },
  {
      "ID": 25,
      "CreatedAt": "2020-06-26T07:52:29.048139Z",
      "UpdatedAt": "2020-06-26T07:52:29.048139Z",
      "SerialNumber": "CH-2",
      "Brand": "CBTL",
      "Status": "New",
      "Model": "Tumblr",
      "DateBought": "Today"
  }
]

describe("Render Search Result tests",() => {
  test("Compare snapshot test",() => {
    const component = renderer.create(
      <SearchResults searchResults={mockSearchResults}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  test("Check for presence of 2 search result cards",() => {
    const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
    expect(component.find(Card).length).toEqual(2)
  })
  mockSearchResults.forEach((result,idx) => {
    test(`Check for presence of result card with brand name ${result["Brand"]}`,() => {
      const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
      expect(component.find(CardContent).at(idx).childAt(0).text()).toEqual("Brand: " + result["Brand"])
    })
    test(`Check for presence of result card with model name ${result["Model"]}`,() => {
      const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
      expect(component.find(CardContent).at(idx).childAt(1).text()).toEqual("Model: " + result["Model"])
    })
    test(`Check for presence of result card with Status name ${result["Status"]}`,() => {
      const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
      expect(component.find(CardContent).at(idx).childAt(2).text()).toEqual("Status: " + result["Status"])
    })
    test(`Check for presence of result card with Serial Number ${result["SerialNumber"]}`,() => {
      const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
      expect(component.find(CardContent).at(idx).childAt(3).text()).toEqual("Serial Number: " + result["SerialNumber"])
    })
    test(`Check for presence of result card with Date Bought ${result["DateBought"]}`,() => {
      const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
      expect(component.find(CardContent).at(idx).childAt(4).text()).toEqual("Date Bought: " + result["DateBought"])
    })
    test(`Check for presence of result card with Updated At ${result["UpdatedAt"]}`,() => {
      const component = shallow(<SearchResults searchResults={mockSearchResults}/>)
      expect(component.find(CardContent).at(idx).childAt(5).text()).toEqual("Updated At: " + result["UpdatedAt"])
    })
  })
})