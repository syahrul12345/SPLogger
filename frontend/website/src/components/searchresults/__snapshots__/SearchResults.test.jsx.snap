// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Search Result tests Compare snapshot test 1`] = `
Array [
  <h5
    className="MuiTypography-root MuiTypography-h5"
  >
     Search Results
  </h5>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Brand: 
            Starbucks
          </h5>
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Model: 
            Cup
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Status: 
            New
          </p>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
          >
            Serial Number: 
            CH-1
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
          >
            Date Bought: 
            Today
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
          >
            Updated At: 
            2020-06-26T07:51:48.306436Z
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Brand: 
            CBTL
          </h5>
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Model: 
            Tumblr
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Status: 
            New
          </p>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
          >
            Serial Number: 
            CH-2
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
          >
            Date Bought: 
            Today
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
          >
            Updated At: 
            2020-06-26T07:52:29.048139Z
          </h6>
        </div>
      </div>
    </div>
  </div>,
]
`;
